ui/app/src/components/RemoteRepositoriesGrid/RemoteRepositoriesGrid.tsx

```diff
@@ -0,0 +1,181 @@
+import { showErrorDialog } from '../../state/reducers/dialogs/error';
+import { useActiveSiteId } from '../../utils/hooks';
+import { useDispatch } from 'react-redux';
+
+export interface RemoteRepositoriesGridProps {
+  resource: Resource<Array<Repository>>;
+  disableActions: boolean;
+  fetchStatus(): void;
+  fetchRepositories(): void;
+}
+
+const messages = defineMessages({
+  mergeStrategyNone: {
+    id: 'repositories.mergeStrategyNone',
+    defaultMessage: 'None'
+  },
+  mergeStrategyOurs: {
+    id: 'repositories.mergeStrategyOurs',
+    defaultMessage: 'Accept Ours'
+  },
+  mergeStrategyTheirs: {
+    id: 'repositories.mergeStrategyTheirs',
+    defaultMessage: 'Accept Theirs'
+  },
+  remoteDeleteSuccessMessage: {
+    id: 'repositories.remoteDeleteSuccessMessage',
+    defaultMessage: 'Remote repository deleted successfully.'
+  },
+  pullSuccessMessage: {
+    id: 'repositories.pullSuccessMessage',
+    defaultMessage: 'Successfully pulled.'
+  },
+  pushSuccessMessage: {
+    id: 'repositories.pushSuccessMessage',
+    defaultMessage: 'Successfully pushed.'
+  }
+});
+
+export default function RemoteRepositoriesGrid(props: RemoteRepositoriesGridProps) {
+  const { resource, disableActions, fetchStatus, fetchRepositories } = props;
+  const [openRepositoriesPullDialog, setOpenRepositoriesPullDialog] = useState(false);
+  const [openRepositoriesPushDialog, setOpenRepositoriesPushDialog] = useState(false);
+  const [repositoriesPullDialogBranches, setRepositoriesPullDialogBranches] = useState([]);
+  const [repositoriesPushDialogBranches, setRepositoriesPushDialogBranches] = useState([]);
+  const [pullRemoteName, setPullRemoteName] = useState(null);
+  const [pushRemoteName, setPushRemoteName] = useState(null);
+  const { formatMessage } = useIntl();
+  const mergeStrategies: MergeStrategy[] = [
+    {
+      key: 'none',
+      value: formatMessage(messages.mergeStrategyNone)
+    },
+    {
+      key: 'ours',
+      value: formatMessage(messages.mergeStrategyOurs)
+    },
+    {
+      key: 'theirs',
+      value: formatMessage(messages.mergeStrategyTheirs)
+    }
+  ];
+  const siteId = useActiveSiteId();
+  const dispatch = useDispatch();
+
+  const onClickPull = (remoteName: string, branches: string[]) => {
+    setRepositoriesPullDialogBranches(branches);
+    setPullRemoteName(remoteName);
+    setOpenRepositoriesPullDialog(true);
+  };
+
+  const onClickPush = (remoteName: string, branches: string[]) => {
+    setRepositoriesPushDialogBranches(branches);
+    setPushRemoteName(remoteName);
+    setOpenRepositoriesPushDialog(true);
+  };
+
+  const onPullSuccess = () => {
+    fetchStatus();
+    setOpenRepositoriesPullDialog(false);
+    dispatch(
+      showSystemNotification({
+        message: formatMessage(messages.pullSuccessMessage)
+      })
+    );
+  };
+
+  const onPullError = (response) => {
+    fetchStatus();
+    setOpenRepositoriesPullDialog(false);
```

Comment:
@rart on pullError/pushError I'm closing the dialog because when there are conflicts the API returns the error, so the dialog goes away and the status component gets updated.